src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -177,19 +185,30 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		if (!decoration) {
 			return undefined;
 		}
+		let color = command.exitCode === undefined ? defaultColor : command.exitCode ? errorColor : successColor;
+		if (color && typeof color !== 'string') {
+			color = color.toString();
+		} else {
+			color = '';
+		}
+
+		this._overviewRuler = this._terminal.registerDecoration({ marker: command.marker, overviewRulerOptions: { color } });
```

Comment:
You should add the options to `decoration` created above instead of creating a new decoration